File: Is Subsequence/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int l1 = s.length();
        int l2 = t.length();
        int j = 0;
        if (l1 > l2)
            return false;
        for (int i = 0; i < l2; i++) {
            if (s[j] == t[i])
                j++;
        }
        if (j == s.length())
            return true;
        return false;
    }
};

File: Is Subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        l1 = len(s)
        l2 = len(t)
        j = 0  

        if l1 > l2:
            return False

        for i in range(l2):
            if j < l1 and s[j] == t[i]:
                j += 1

        return j == l1
